java/src/main/native/ai_onnxruntime_OrtSession_SessionOptions.c

```diff
@@ -615,3 +616,50 @@ JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_00024SessionOptions_addROC
     throwOrtException(jniEnv,convertErrorCode(ORT_INVALID_ARGUMENT),"This binary was not compiled with ROCM support.");
   #endif
 }
+
+/*
+ * Class::    ai_onnxruntime_OrtSession_SessionOptions
+ * Method:    appendExecutionProvider
+ * Signature: (JILjava/lang/String)V
+ */
+JNIEXPORT void JNICALL Java_ai_onnxruntime_OrtSession_00024SessionOptions_sessionOptionsAppendExecutionProvider(
+    JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong optionsHandle,
+    jstring jepName, jobjectArray configKeyArr, jobjectArray configValueArr) {
+  (void)jobj;
+  char errMsgBuf[512];
+  const char* epName = (*jniEnv)->GetStringUTFChars(jniEnv, jepName, NULL);
+  if (strcmp(epName, "XNNPACK") == 0) {
+#ifndef USE_XNNPACK
+    throwOrtException(jniEnv, convertErrorCode(ORT_INVALID_ARGUMENT), "This binary was not compiled with Xnnpack support.");
+#endif
+  } else {
+    snprintf(errMsgBuf, 512, "Java Interface doesn't support this EP %s", epName);
+    throwOrtException(jniEnv, convertErrorCode(ORT_INVALID_ARGUMENT), errMsgBuf);
+  }
+
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtSessionOptions* options = (OrtSessionOptions*)optionsHandle;
+  #define  MAX_CONFIG_SIZE  20
```

Comment:
is it necessary to have a hard coded limit?
if we need to use a macro, can we `#undef` it after it is used?

Reply:
I think I'd prefer to malloc something here rather than have a constant sized array. If we have a constant sized array then the Java code should do the max length check as it's easier to catch & deal with there.

Reply:
At the begining, I used the `size of configKeyArr`, however  something like 
```
int sizeOfKey = xx.size();
const char*  cKeyArray[sizeOfKey ].
```
will be failed to compile under winidows. `constant expression` expected.

Would a static stack array be easier to manager than dynamic allocated memory?